homeassistant/helpers/reload.py

```diff
@@ -98,6 +99,19 @@ async def _resetup_platform(
         await _async_reconfig_platform(platform, root_config[integration_platform])
         return
 
+    # if still empty, config might be new adr0007 style.
+    if not root_config[integration_platform] and conf.get(integration_name):
```

Comment:
What if they have both old and new style?

Reply:
Only old style will be used in that case. That's because we still have the old-old-style config for group integration (https://www.home-assistant.io/integrations/group#old-style-groups), which is in conflict with this ADR0007 new style configuration. If you want we could include an exception for this special integration, but I would not like it because than the code will contain a 'group-integration-specific-exception'.... Alternatively, we could drop that old-style group config. In that case we can simply add all new-style configs to the root-config, without checking if it is empty or not in advance.